flexmeasures/ui/templates/crud/asset.html

```diff
@@ -798,37 +981,326 @@ <h5 class="card-title">${sensor.name}</h5>
+                assetFlexContext[flexId].push(sensorId);
+            } else {
+                assetFlexContext[flexId] = { sensor: sensorId };
+            }
+        }
+        highlightedCard = await renderFlexContextInputField(flexId);
+        flexContextFormEle.innerHTML = flexContextFormEle.innerHTML.replace(document.getElementById(`${flexId}-control`).outerHTML, highlightedCard);
+        await selectFlexField(document.getElementById(`${flexId}-control`));
+    }
+
+    async function addFlexContextField() {
+        const fieldName = flexSelect.value;
+        if (fieldName === "blank") {
+            showToast("Please select a field to add", "error");
+            return;
+        }
+        assetFlexContext[fieldName] = "";
+        flexOptionsContainer.innerHTML = "";
+        await renderFlexContextForm(); // im rendring here so te card with the 'border-on-click' class is updated/exists
+        const card = document.getElementById(`${fieldName}-control`);
+        selectFlexField(card);
+    }
+
+    function renderFlexSelectOptions() {
+        flexSelect.innerHTML = `<option value="blank">Select an option</option>`;
+        for (const [key, value] of Object.entries(assetFlexContext)) {
+            if (value === null) {
+                flexSelect.innerHTML += `<option value="${key}">${key.split("-").map((word) => word.charAt(0).toUpperCase() + word.slice(1)).join(" ")}</option>`;
+            }
+        }
+
+    }
+
+    function renderFlexInputOptions(contextKey, isArray = false) {
+        return `
+                <div class="card-header">
+                    <ul class="nav nav-tabs card-header-tabs" id="pills-tab" role="tablist">
+                        <li class="nav-item" role="presentation">
+                            <button class="nav-link active fixed-val" id="pills-homfixed" data-bs-toggle="pill" data-bs-target="#pills-fixed" type="button" role="tab" aria-controls="pills-fixed" aria-selected="true">
+                                Fixed Value
+                                <i class="fa fa-info-circle ps-2"
+                                    data-bs-toggle="tooltip"
+                                    data-bs-placement="bottom"
+                                    title="Some Info about this field"
+                                ></i>
+                            </button>
+                        </li>
+                        <li class="nav-item" role="presentation">
+                            <button class="nav-link dynamic-val" id="pills-dynamic-tab" data-bs-toggle="pill" data-bs-target="#pills-dynamic" type="button" role="tab" aria-controls="pills-dynamic" aria-selected="false">
+                                Dynamic Value(a Sensor)
+                                <i class="fa fa-info-circle ps-2"
+                                    data-bs-toggle="tooltip"
+                                    data-bs-placement="bottom"
+                                    title="Some Info about this field"
```

Comment:
"You can a sensor which provides the values in a dynamic fashion. In the search form below, you can choose the right one. Be sure the data will be available consistently, though, so scheduling will work."